anywidget/_descriptor.py

```diff
@@ -521,7 +523,10 @@ def determine_state_setter(obj: object) -> Callable[[object, dict], None]:
 
 def _get_psygnal_signal_group(obj: object) -> psygnal.SignalGroup | None:
     """Look for a psygnal.SignalGroup on the obj."""
-    psygnal = sys.modules.get("psygnal")
+    if TYPE_CHECKING:
+        import psygnal
+    else:
+        psygnal = sys.modules.get("psygnal")
```

Comment:
no, I don't think there is, that's much better :joy:

Reply:
actually, never mind... i guess i did it for a reason, and that is because, apparently, when you redefine a name within a function scope, it looks like the globals type doesn't carry in (and it shouldn't!), so if I don't do that, you'll see this:

<img width="458" alt="Screen Shot 2023-02-15 at 6 15 08 PM" src="https://user-images.githubusercontent.com/1609449/219213663-f90a1a55-b171-4f97-9505-2c8c305e57c2.png">

<img width="813" alt="Screen Shot 2023-02-15 at 6 15 14 PM" src="https://user-images.githubusercontent.com/1609449/219213699-b319171c-8f27-4efe-9195-06ada35cd201.png">

of course, it does come at the cost of a runtime conditional evaluation (negligible), and the alternative would be a `type: ignore`

Reply:
Ahh, ok. Let's keep it as is, not worried about the basically non-runtime impact :)